### PR TITLE
Add roles-settings argument to hc sandbox

### PR DESCRIPTION
**Pull request by matthme**
### Summary

Addresses another part of https://github.com/holochain/holochain/issues/3918 by exposing the new `roles_settings` option intriduced with https://github.com/holochain/holochain/pull/4416 in hc sandbox as well.

### TODO:
- [ ] CHANGELOGs updated with appropriate info
- [ ] All code changes are reflected in docs, including module-level docs

### CODE REVIEW

File: crates/hc_sandbox/tests/cli.rs [hunk 1]
```diff
@@ -273,6 +274,127 @@ async fn generate_sandbox_memproof_deferred_and_call_list_dna() {
     assert!(exit_code.success());
 }
 
+/// Generates a new sandbox with roles settings overriden by a yaml file passed via
```
Comment (jost-s) [inline crates/hc_sandbox/tests/cli.rs:277]:
```suggestion
/// Generates a new sandbox with roles settings overridden by a yaml file passed via
```

File: CHANGELOG.md
Comment (jost-s) [inline CHANGELOG.md:1]: Looks like there's some formatter at work =) Please undo the changes after your addition.

**Review by jost-s — CHANGES_REQUESTED** [review]
Feature LGTM

**Comment by ThetaSinner** [comment]
I think this test is genuinely flaky and not just flaky on one platform - I've had a go at improving it here https://github.com/holochain/holochain/pull/4457

File: crates/holochain/CHANGELOG.md
[the remark this answers is not in the document]
Comment (matthme) [reply crates/holochain/CHANGELOG.md:1]: I didn't figure out where that formatter setting would be...but I reverted the formatting anyway, at least I learnt how to save without formatting now. In what regard would these formatting changes have been a problem? In that they cause merge conflicts for other people that currently have active branches not in develop?

[the remark this answers is not in the document]
Comment (jost-s) [reply crates/holochain/CHANGELOG.md:1]: In VSCodium in the settings json you can specify
```json
"[markdown]": {"editor.formatOnSave": false},
```
in case you're using it.

The problem is generally that there are diffs that don't belong to what the PR is about. And if different people use different formatters, the formatting is changed around with every PR.

**Review by jost-s — APPROVED** [review]

File: crates/holochain/src/conductor/conductor.rs [hunk 2]
```diff
@@ -1436,6 +1436,8 @@ pub struct InstallAppCommonFlags {
 }
 
 /// Methods related to app installation and management
+///
+/// Tests related to app installation can be found in ../../test/tests/app_installation/mod.rs
```
Comment (c12i) [inline crates/holochain/src/conductor/conductor.rs:1440]:
```suggestion
/// Tests related to app installation can be found in ../../tests/tests/app_installation/mod.rs
```

File: crates/hc_sandbox/src/calls.rs [hunk 3]
```diff
@@ -540,13 +547,22 @@ pub async fn install_app_bundle(cmd: &mut CmdRunner, args: InstallApp) -> anyhow
...
+        roles_settings,
     } = args;
 
+    let roles_settings = match roles_settings {
+        None => None,
+        Some(path) => {
+            let yaml_string = std::fs::read_to_string(path)?;
+            Some(serde_yaml::from_str::<RoleSettingsMap>(&yaml_string)?)
+        }
+    };
```
Comment (c12i) [inline crates/hc_sandbox/src/calls.rs:559]: Minor, but more idiomatic to start matching `Some` before `None`

```suggestion
    let roles_settings = match roles_settings {
        Some(path) => {
            let yaml_string = std::fs::read_to_string(path)?;
            Some(serde_yaml::from_str::<RoleSettingsMap>(&yaml_string)?)
        },
         None => None,
    };
```

File: crates/holochain/src/conductor/conductor.rs (hunk 2 above)
Comment (matthme) [reply crates/holochain/src/conductor/conductor.rs]: thanks, resolved.

File: crates/hc_sandbox/src/calls.rs (hunk 3 above)
Comment (matthme) [reply crates/hc_sandbox/src/calls.rs]: addressed :+1:
